sale_promotion_rule/models/sale_promotion_rule.py

```diff
@@ -0,0 +1,317 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018  Acsone SA/NV (http://www.acsone.eu)
+# Copyright 2017 Akretion (http://www.akretion.com).
+# @author Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+from collections import defaultdict
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare
+import odoo.addons.decimal_precision as dp
+
+_logger = logging.getLogger(__name__)
+
+
+class SalePromotionRule(models.Model):
+    _name = 'sale.promotion.rule'
+    _description = 'Sale Promotion Rule'
+
+    sequence = fields.Integer(default=10)
+    rule_type = fields.Selection(
+        selection=[
+            ('coupon', 'Coupon'),
+            ('auto', 'Automatic'),
+            ],
+        required=True,
+        default='coupon')
+    name = fields.Char(required=True,
+                       translate=True)
+    code = fields.Char()
+    discount_amount = fields.Float(
+        digits=dp.get_precision('Discount'),
+        required=True)
+    promo_type = fields.Selection(
+        selection=[
+            # ('gift', 'Gift'), TODO implement
+            ('discount', 'Discount'),
+            ],
+        required=True,
+        default='discount')
+    discount_type = fields.Selection(
+        selection=[
+            ('percentage', 'Percentage'),
+            # ('amount', 'Amount'), TODO implement
+            ],
+        required=True,
+        default='percentage')
+    date_from = fields.Date()
+    date_to = fields.Date()
+    only_newsletter = fields.Boolean()
+    restrict_partner_ids = fields.Many2many(
+        comodel_name='res.partner',
+        relation='discount_rule_partner_rel',
+        column1='rule_id',
+        column2='partner_id',
+        string='Restricted partners')
+    restrict_pricelist_ids = fields.Many2many(
+        comodel_name='product.pricelist',
+        relation='discount_rule_pricelist_rel',
+        column1='rule_id',
+        column2='pricelist_id',
+        string='Restricted pricelists')
+    usage_restriction = fields.Selection(
+        selection=[
+            ('one_per_partner', 'One per partner'),
+            ('no_restriction', 'No restriction')
+        ], default='no_restriction',
+        required=True)
+    restriction_amount = fields.Selection(
```

Comment:
IMHO the name is quite weird as we are waiting for a float. Maybe 'restriction_amount_filed ?

Reply:
done